ferrocene/doc/requirements/src/self-test.rst

```diff
@@ -0,0 +1,209 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Ferrocene Developers
+
+`ferrocene-self-test` tool
+==========================
+
+General checks
+--------------
+
+Checks for executables
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+
+   **executable** refers to the following:
+
+   - ``rustc``
+   - ``rustdoc``
+
+The executable should exist in the ``bin`` directory, relative to the root of the installation.
+
+The metadata of the executable should be fetchable.
+
+The executable should be a regular file.
+
+The executable should have read and execute permissions set for all users.
+
+When executed with ``-vV`` command-line options, the following apply:
+
+- The executable should exit successfully.
+
+- The output of the executable should be UTF-8 text.
+
+- The output of the executable should have at least 3 values: host, commit-hash, and release.
+
+- The value of ``host`` from the output of the executable should match the target-triple
+  of the executing environment.
+
+- The value of ``commit-hash`` from the output of the executable should match the 
+  commit hash of the Ferrocene release corresponding to the installed Ferrocene toolchain.
+
+- The value of ``release`` from the output of the executable should match the
+  version of the Ferrocene release corresponding to the installed Ferrocene toolchain.
+
+Checks for targets
+------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Target
+     - Has a standard library (``std``)
+   * - :target-with-triple:`x86_64-unknown-linux-gnu`
+     - ✔
+   * - :target-with-triple:`aarch64-unknown-none`
+     -
+   * - :target-with-triple:`x86_64-pc-windows-msvc`
+     - ✔
+   * - :target-with-triple:`aarch64-unknown-linux-gnu`
+     - ✔
+   * - :target-with-triple:`aarch64-apple-darwin`
+     - ✔
+   * - :target-with-triple:`x86_64-apple-darwin`
+     - ✔
+   * - :target-with-triple:`aarch64-unknown-nto-qnx710`
+     - ✔
+   * - :target-with-triple:`x86_64-pc-nto-qnx710`
+     - ✔
+   * - :target-with-triple:`armv8r-none-eabihf`
+     -
+   * - :target-with-triple:`thumbv7em-none-eabi`
+     -
+   * - :target-with-triple:`thumbv7em-none-eabihf`
+     -
+   * - :target-with-triple:`armv7r-none-eabihf`
+     -
+   * - :target-with-triple:`armebv7r-none-eabihf`
+     -
```

Comment:
This list will get out of date. We should instead refer to the target support list in the user manual (and mentions what to look for in that page).

Reply:
5e27d0bea76